### PR TITLE
net:fix coverity warning

### PR DESCRIPTION
**Pull request by hjlinux**
Signed-off-by: hejianliang3 <[hejianliang3@xiaomi.com](mailto:hejianliang3@xiaomi.com)>

### CODE REVIEW

File: net/socket/connect.c [hunk 1]
```diff
@@ -140,7 +140,7 @@ int psock_connect(FAR struct socket *psock, FAR const struct sockaddr *addr,
   if (ret >= 0)
     {
       FAR struct socket_conn_s *conn = psock->s_conn;
-      conn->s_flags |= _SF_CONNECTED;
```
Comment (acassis) [inline net/socket/connect.c]: Strange, checkpatch.sh didn't find this issue:
$ ./tools/checkpatch.sh -f net/rpmsg/rpmsg_sockif.c

There should be some BUG in the nxstyle.c

**Comment by pkarashchenko** [comment]
`Remove Gerrit Change-ID's before submitting upstream`

File: net/socket/connect.c (hunk 1 above)
Comment (xiaoxiang781216) [reply net/socket/connect.c]: nxstyle is hard to find the check duplication.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by masayuki2009** [comment]
@hjlinux @xiaoxiang781216 
Hmm, I noticed that my automatic test failed with lm3s9665-ek:discover (QEMU)
It seems that telnetd on NuttX does not work correctly.
If I revert this PR, it works correctly.

```
$ ~//opensource/QEMU/qemu-5.2/build/arm-softmmu/qemu-system-arm -net nic,model=stellaris -net user,hostfwd=tcp:127.0.0.1:20023-10.0.2.15:23,hostfwd=tcp:127.0.0.1:20021-10.0.2.15:21,hostfwd=tcp:127.0.0.1:25001-10.0.2.15:5001 -M lm3s6965evb -kernel ./nuttx/nuttx -nographic

NuttShell (NSH) NuttX-10.3.0-RC0
nsh> uname -a
NuttX  10.3.0-RC0 e9648d8a73 Apr  3 2022 22:13:08 arm lm3s6965-ek
nsh> ps
  PID GROUP PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0 FIFO     Kthread N-- Ready              00000000 001000 000384  38.4%  Idle Task
    1     1 224 RR       Kthread --- Waiting  Semaphore 00000000 002016 000260  12.8%  hpwork 0x20000624
    2     2  60 RR       Kthread --- Waiting  Semaphore 00000000 002016 000260  12.8%  lpwork 0x20000634
    3     3 100 RR       Task    --- Running            00000000 002024 001192  58.8%  nsh_main
    4     4 100 RR       Task    --- Waiting  Semaphore 00000000 002008 000492  24.5%  Telnet daemon 0x200063e8
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:      50544      14992      35552      35552     53      1
nsh> ifconfig
eth0	Link encap:Ethernet HWaddr 00:e0:de:ad:be:ef at UP
	inet addr:10.0.0.2 DRaddr:10.0.0.1 Mask:255.255.255.0

             IPv4   TCP   UDP  ICMP
Received     0000  0000  0000  0000
Dropped      0000  0000  0000  0000
  IPv4        VHL: 0000   Frg: 0000
  Checksum   0000  0000  0000  ----
  TCP         ACK: 0000   SYN: 0000
              RST: 0000  0000
  Type       0000  ----  ----  0000
Sent         0000  0000  0000  0000
  Rexmit     ----  0000  ----  ----
nsh> renew eth0
nsh> ifconfig
eth0	Link encap:Ethernet HWaddr 00:e0:de:ad:be:ef at UP
	inet addr:10.0.2.15 DRaddr:10.0.2.2 Mask:255.255.255.0

             IPv4   TCP   UDP  ICMP
Received     0002  0000  0002  0000
Dropped      0000  0000  0000  0000
  IPv4        VHL: 0000   Frg: 0000
  Checksum   0000  0000  0000  ----
  TCP         ACK: 0000   SYN: 0000
              RST: 0000  0000
  Type       0000  ----  ----  0000
Sent         0002  0000  0002  0000
  Rexmit     ----  0000  ----  ----
nsh> 

Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Traceback (most recent call last):
  File "/home/ishikawa/script/expect_nuttx_dhcpc_telnetd_test.py", line 46, in <module>
    child2.expect('nsh> ', timeout=5)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 344, in expect
    timeout, searchwindowsize, async_)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/spawnbase.py", line 372, in expect_list
    return exp.expect_loop(timeout)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
  File "/home/ishikawa/.local/lib/python3.6/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
```

**Comment by xiaoxiang781216** [comment]
Could you try restore this line?
image
The code want to compute the checksum of src and dest ip,  but Coverity mistakenly think that the code access out of src ip memory.

**Comment by masayuki2009** [comment]
>The code want to compute the checksum of src and dest ip, but Coverity mistakenly think that the code access out of src ip memory.

@xiaoxiang781216 @hjlinux 
The change works but I think you should do testing before sending the PR.

**Comment by xiaoxiang781216** [comment]
@hjlinux please do the testing next time.

**Comment by hjlinux** [comment]
@xiaoxiang781216  @masayuki2009  
ok, I'll do the test before PR

**Pull request merged**